bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyDeviceProfile.java

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.shelly.internal.api;
+
+import static org.openhab.binding.shelly.internal.ShellyBindingConstants.SHELLYDT_DIMMER;
+import static org.openhab.binding.shelly.internal.ShellyUtils.*;
+import static org.openhab.binding.shelly.internal.api.ShellyApiJson.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang.Validate;
+import org.openhab.binding.shelly.internal.ShellyBindingConstants;
+import org.openhab.binding.shelly.internal.ShellyUtils;
+import org.openhab.binding.shelly.internal.api.ShellyApiJson.ShellySettingsGlobal;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link ShellyDeviceProfile} creates a device profile based on the settings returned from the API's /settings
+ * call. This is used to be more dynamic in controlling the device, but also to overcome some issues in the API (e.g.
+ * RGBW2 returns "no meter" even it has one)
+ *
+ * @author Markus Michels - Initial contribution
+ */
+public class ShellyDeviceProfile {
```

Comment:
Not directly. This means that I need to tag all members as @nullable or initialize. Let's keep this on the list after the release. I don't want to make cosmetical changes before RC1. Some of the members might become null, because the device doesn't report those (depending on the device type) and that could lead into a wrong null check.